db/migrate/20170531140414_create_federation_config.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+class CreateFederationConfig < ActiveRecord::Migration
+  def self.up
+    create_table :federation_configs do |t|
+      t.integer :collection_id, null: false # The ID of the collection this config applies to
+      t.string :provider, null: false       # The name of the provider this config applies to
+      t.string :context_query               # The default query which should always be applied.
```

Comment:
generally yes, however I'd disagree on this since "query" is already used by both the portal and the foederati gem to refer to the search term/the final search string. I think it's nicer to differentiate this from the rest of the "query" as it is only a partial string of the final "query".